### ISSUE TITLE
Feature Request Web Link Pop-Up

### ISSUE THREAD

**Issue by PaltryProgrammer** [notepad-plus-plus/notepad-plus-plus#12209]
Greetings and Kind Regards BowPad has a nice feature to wit i.e. upon hovering over a web link a pop-up box appears upon which can be clicked upon which then transports to the web page as per usual. The pop-up box also reminds which mouse click key combination to utilize to perform same if clicking on the link instead of the pop-up box. I find these very helpful as I do not care for combinations and as each editor requires a different such combination making memory of each uncertain and unreliable so becomes a nuisance as then one must attempt every possible combination to wit i.e. Ctrl+click , Shift+click , Alt+click or double click until one discovers the correct combination for any particular editor - Kind Regards - Cheerio
Please see attached GIF for demonstration of said web link pop-up
![Request Web Link Pop-Up](https://user-images.githubusercontent.com/48535920/191642897-a5ebe6b8-7124-4cb0-bc9b-90124b122b9a.gif)

**Comment by alankilborn** [comment]
Personally, I don't like it, as it just would provide more visual clutter/distraction for me.
If it were optional I suppose it would be fine.
Perhaps something can be scripted to provide the functionality.

**Comment by alankilborn** [comment]
This PythonScript can be used as a workaround, or as a simple demo showing it actually working in Notepad++:

```
# -*- coding: utf-8 -*-
from __future__ import print_function

# references:
#  https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12209 "Feature Request Web Link Pop-Up"

from Npp import *

#-------------------------------------------------------------------------------

class UHT(object):

    def __init__(self):
        self.debug = True if 0 else False
        self.URL_INDIC = 8  # URL_INDIC is used in N++ source code
        editor.setMouseDwellTime(200)  # the less the value the greater the chance to get false info
        editor.callback(self.on_mouse_dwell_start_callback, [SCINTILLANOTIFICATION.DWELLSTART])

    def on_mouse_dwell_start_callback(self, args):
        self.print('DWELLSTART:', args)
        self.dwell_pos = editor.charPositionFromPointClose(args['x'], args['y'])
        (start_pos, end_pos) = self.get_indicator_range(self.URL_INDIC, self.dwell_pos)
        self.print('start_pos:', start_pos, 'end_pos:', end_pos)
        if start_pos != end_pos:
            self.call_tip_display('Double-click to follow link; single-click to edit')
        else:
            if editor.callTipActive(): editor.callTipCancel()

    def call_tip_display(self, text, above=False):
        if editor.callTipActive(): editor.callTipCancel()
        #editor.callTipSetPosition(False)  # leave this False because the N++ regex search tooltip will be affected
        editor.callTipShow(self.dwell_pos, text)

    def get_indicator_range(self, indic_number, position=None):
        # similar to ScintillaEditView::getIndicatorRange() in N++ source
        # https://github.com/notepad-plus-plus/notepad-plus-plus/blob/8f38707d33d869a5b8f5014dbb18619b166486a0/PowerEditor/src/ScitillaComponent/ScintillaEditView.h#L562
        curr_pos = editor.getCurrentPos() if position == None else position
        indic_mask = editor.indicatorAllOnFor(curr_pos)
        if (indic_mask & (1 << indic_number)) != 0:
            start_pos = editor.indicatorStart(indic_number, curr_pos)
            end_pos = editor.indicatorEnd(indic_number, curr_pos)
            if curr_pos >= start_pos and curr_pos <= end_pos:
                return (start_pos, end_pos)
        return (0, 0)

    def print(self, *args):
        if self.debug:
            #console.show()
            print('UHT:', *args)

#-------------------------------------------------------------------------------

# to run via another file, e.g., startup.py, put these lines (uncommented and unindented) in that file:
#  import UrlHoverTip
#  uht = UrlHoverTip.UHT()

if __name__ == '__main__': UHT()
```

Here's a screenshot of it working:

![image](https://user-images.githubusercontent.com/77065706/191749402-70fcfc78-ddb7-416a-9bda-24cc3360a4de.png)

**Comment by alankilborn** [comment]
>Personally, I don't like it, as it just would provide more visual clutter/distraction for me.

And even while running the provided script to just experiment with it, it did indeed do that.  The pop up hover tip annoys me...a lot.

**Comment by alankilborn** [comment]
Another user has requested this on the Community: https://community.notepad-plus-plus.org/topic/26041

**Comment by pryrt** [comment]
Closed as duplicate of #10650
